gateway/gateway.go

```diff
@@ -376,10 +379,24 @@ func (gateway *HandleT) userWebRequestWorkerProcess(userWebRequestWorker *userWe
 			// set anonymousId if not set in payload
 			var index int
 			result := gjson.GetBytes(body, "batch")
-			newAnonymousID := uuid.NewV4().String()
+
 			var reqMessageIDs []string
+			var notIdentifiable bool
 			result.ForEach(func(_, _ gjson.Result) bool {
-				if strings.TrimSpace(gjson.GetBytes(body, fmt.Sprintf(`batch.%v.anonymousId`, index)).String()) == "" {
+				anonIDFromReq := strings.TrimSpace(gjson.GetBytes(body, fmt.Sprintf(`batch.%v.anonymousId`, index)).String())
```

Comment:
If the request already has an anonymousId - we can reuse it for all other events in the batch.
Similarly, if we have userID in one of the events, we should reuse the same hash as anonymousID.

Basically get these anonymousID out of the for loop and use the same across all events in the loop. Thoughts?